multicore_runtime/wsgi_config.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Utilities to configure the dispatcher, middleware and environment."""
+
+import logging
+import os
+import threading
+import UserDict
+
+from static_files import static_app_for_regex_and_files
+
+from google.appengine.api import appinfo
+from google.appengine.api import appinfo_includes
+from google.appengine.runtime import wsgi
+
+
+def get_module_config_filename():
+  """Returns the name of the module configuration file (app.yaml)."""
+  module_yaml_path = os.environ['MODULE_YAML_PATH']
+  logging.info('Using module_yaml_path from env: %s', module_yaml_path)
+  return module_yaml_path
+
+
+def get_module_config(filename):
+  """Returns the parsed module config."""
+  with open(filename) as f:
+    return appinfo_includes.Parse(f)
+
+
+def app_for_script(script):
+  """Returns the WSGI app specified in the input string, or None on failure."""
+  try:
+    app, filename, err = wsgi.LoadObject(script)  # pylint: disable=unused-variable
+  except ImportError as e:
+    # Despite nominally returning an error object, LoadObject will sometimes
+    # just result in an exception. Since we're already processing the err
+    # object, we might as well just use that variable to store the exception.
+    err = e
+  if err:
+    # Log the exception but do not reraise.
+    logging.exception('Failed to import %s: %s', script, err)
```

Comment:
The `logging.exception` command only works when inside an `except` block.

I think reusing `err` for the `LoadObject` error and for a possible exception might not be the best approach here, I think having separate logging lines for each would make sense (you might possibly want to differentiate them, right?) and repeating the `return None` is probably OK as it's simple enough.

Alternatively, if you want to have a single `logging` line, then use `logging.error(...)` and set the `exc_info=...` named parameter to either the exception object (`e` here, but find a better name if you want to keep it around) or to `None` if no exception happened. You might want to set `err` to something else in that case, maybe a string with a message saying it failed due to a Python exception (and then the `exc_info being set will include the exception itself with the log?)

How does that sound?